src/ansys/fluent/core/launcher/watchdog.py

```diff
@@ -0,0 +1,302 @@
+import multiprocessing.pool
+import os
+from pathlib import Path
+import random
+import string
+import subprocess
+import sys
+import time
+
+import ansys.fluent.core as pyfluent
+
+IDLE_PERIOD = 2  # seconds
+WATCHDOG_INIT_FILE = "watchdog_{}_init"
+
+
+def launch(main_pid: int, sv_port: int, sv_password: str, sv_ip: str = None):
+    watchdog_id = "".join(
+        random.choices(
+            string.ascii_uppercase + string.ascii_lowercase + string.digits, k=6
+        )
+    )
+
+    env_watchdog_debug = os.getenv("PYFLUENT_WATCHDOG_DEBUG", "off").upper()
+    if env_watchdog_debug in [1, "1", "ON"]:
+        print(f"Number of arguments: {len(sys.argv)} arguments.")
+        print(f"Argument list: {sys.argv}")
+
+        print(
+            f"PYFLUENT_WATCHDOG_DEBUG environment variable found, "
+            f"enabling debugging for watchdog id {watchdog_id}..."
+        )
+
+    logger = pyfluent.logging.get_logger("pyfluent.launcher")
+
+    watchdog_env = os.environ.copy()
+
+    # No auto PyFluent logging to file on the watchdog
```

Comment:
@raph-luc is the following code producing the required behaviour?

If the var is not in env, it will still not be in env
If it is in env and falsy, it will still be in env and falsy
If it is in env and truthy, it will not be in env